src/main/java/core/basesyntax/operation/SupplyHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class SupplyHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;

```

Comment:
According to SOLID principles, it's better to depend on abstractions, not on concrete classes. Consider changing `ShopServiceImpl` to its interface.